go/pkg/cas/upload.go

```diff
@@ -935,13 +1063,26 @@ const pathSep = string(filepath.Separator)
 // joinFilePathsFast is a faster version of filepath.Join because it does not
 // call filepath.Clean. Assumes arguments are clean.
 func joinFilePathsFast(a, b string) string {
+	if b == "." {
+		return a
+	}
 	if strings.HasSuffix(a, pathSep) {
 		// May happen if a is the root.
 		return a + b
 	}
 	return a + pathSep + b
 }
 
+// dirNameRelFast is a faster version of filepath.Dir because it does not call
+// filepath.Clean. Assumes the argument is clean and relative.
+func dirNameRelFast(relPath string) string {
```

Comment:
is this technically just `dirNameFast`? IIUC there's no requirement for path to be relative to anything.

Reply:
it does not work for "/" and "C:/"

Reply:
added. Kind expected that "Assumes the argument is clean and relative." is enough